erpnext/stock/doctype/stock_reconciliation/stock_reconciliation.py

```diff
@@ -45,6 +45,7 @@ def on_submit(self):
 	def on_cancel(self):
 		self.delete_and_repost_sle()
 		self.make_gl_entries_on_cancel()
+		self.delete_auto_created_batches()
```

Comment:
Instead of delete should we relink? because there might be chances that batch has used in another transactions after stock reco

Reply:
@rohitwaghchaure i tested this scenario too, if the batch is in another transaction it doesnt let you cancel the **origin** transaction. It throws a "Cannot delete batch as it is linked to another SLE"
I'll work on a test case for these now